src/gretel_synthetics/batch.py

```diff
@@ -579,20 +589,20 @@ def generate_batch_lines(
                 if line.valid is None or line.valid is True:
                     batch.add_valid_data(line)
                     t.update(1)
-                    n_valid += 1
+                    summary.valid_lines += 1
                 else:
                     t2.update(1)
                     batch.gen_data_invalid.append(line)
-                    n_invalid += 1
+                    summary.invalid_lines += 1
         except TooManyInvalidError:
             if raise_on_exceed_invalid:
                 raise
             else:
-                return False
+                return summary
         t.close()
         t2.close()
-        is_valid = batch.gen_data_count >= num_lines
-        return {'valid_lines': n_valid, 'invalid_lines': n_invalid, 'is_valid': is_valid}
+        summary.is_valid = batch.gen_data_count >= num_lines
+        return summary
```

Comment:
This changes the return type in the type hint, you can still return a `dict` if you want, but you have to do something like `summary.__dict__`, otherwise, update the type hint to be clear you're returning a `GenerationSummary` instance

Reply:
I think it's best to return `GenerationSummary`. On the callsite we can change it to whatever is necessary.